pallets/tendermint-light-client/src/lib.rs

```diff
@@ -272,20 +286,29 @@ pub mod pallet {
         }
 
         // TODO: This method will need to be called by the pallet itself if it detects a fork.
+        // TODO : weight depends on whether this is a no-op or not
         /// Halt or resume all light client operations
         ///
         /// Can only be called by root
         #[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
         pub fn set_halted(origin: OriginFor<T>, halted: bool) -> DispatchResult {
             ensure_root(origin)?;
-            <IsHalted<T>>::put(halted);
-
-            if halted {
-                log::warn!(target: "runtime::tendermint-lc", "Halting light client operations");
-                Self::deposit_event(Event::LightClientHalted);
-            } else {
-                log::warn!(target: "runtime::tendermint-lc", "Resuming light client operations");
-                Self::deposit_event(Event::LightClientResumed);
+
+            let current_status = Self::is_halted();
+
+            match [halted, current_status].as_slice() {
```

Comment:
I would just do
```rust

let current_status = Self::is_halted();

if current_status == halted {
  // exit early
  return
}

// old code
```

Seems simpler and I don't think we need to emit any `NoOp` event.

Reply:
no problem, will add these changes in next PRs